src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -168,4 +172,37 @@ public int GetHashCode(JobDefinition obj) => HashCode.Combine(
             obj.CustomName,
             obj.IsStartupJob);
     }
+
+    private sealed class DependentJobDefinitionEqualityComparer : IEqualityComparer<JobDefinition>
+    {
+        // TODO: Maybe is the code conflating two different concepts.
```

Comment:
@linkdotnet Thoughts?

Reply:
DependentJobDefinition - that is a good question. And I can't answer the question until now.

We need a definition of a dependent job and when it gets triggered. 
Will have a look in more depth later today

Reply:
We can invest some time into a new type `DependentJobDefinition`. It feels like it might be useful.
But not in this PR :D